### ISSUE TITLE
When `pages dev` command crashed w/ Functions, spawned process is not killed

### ISSUE THREAD

**Issue by leader22** [cloudflare/wrangler2#83]
## Steps to Reproduce

Prepare static site and Functions.

```
public/
  index.html
functions/
  api.js
```

```js
// functions/api.js
export const onRequestGet = async () => {
  // This is intentional, assuming the exception is not handled
  if (Math.random() < 0.4) throw new Error("OMG");

  return new Response("Hello, world!");
};
```

Then, run the command.

```
npx wrangler@beta pages dev public
```

Check processes at this point.

```
$ ps aux | grep node
leader22         77548   0.0  0.1 34407168  23920 s001  S+   12:16PM   0:00.07 node /Users/leader22/Codes/cfp-repro/node_modules/.bin/wrangler pages dev public
leader22         77549   1.3  0.5 55590896  75536 s001  S+   12:16PM   0:00.73 /Users/leader22/.nodebrew/node/v16.13.0/bin/node --no-warnings --experimental-vm-modules /Users/leader22/Codes/cfp-repro/node_modules/wrangler/wrangler-dist/cli.js pages dev public
leader22         77551   0.9  0.2 34454992  39972 s001  S+   12:16PM   0:00.26 node /Users/leader22/Codes/cfp-repro/node_modules/.bin/serve public
```

Then the API is accessed and an exception is thrown, the main command will be crashed and terminated.

Again, check the process.

```
$ ps aux | grep node
leader22         77551   0.0  0.2 34453968  38236 s001  S    12:16PM   0:00.28 node /Users/leader22/Codes/cfp-repro/node_modules/.bin/serve public
```

This process should be killed too?(Or originally it should not crash...?)

**Comment by threepointone** [comment]
Good catch, this is a bug on our part. Same issue as https://github.com/cloudflare/wrangler2/issues/94, but I'll leave both issues open.

**Comment by GregBrimble** [comment]
Fixed in #105 . We'll hopefully get it released soon :)

**Comment by GregBrimble** [comment]
The server should no longer crash with an error in your functions. But it should also more gracefully exit the child processes now as well.
